.github/workflows/deploy-stg-enhanced-worker.yml

```diff
@@ -0,0 +1,32 @@
+on:
+  push:
+    paths: ["src/**"]
+    branches:
+      - "master"
+      - "integration"
+      - 'DOP-3833'
+
+concurrency:
+  group: environment-stg-${{ github.ref }}
+  cancel-in-progress: true
+name: Deploy Staging ECS
```

Comment:
It might also be worth giving these workflows different names to help distinguish them when filtering between different GH Actions in the UI. Notice the 2 labels for "Deploy Staging ECS" on [this page](https://github.com/mongodb/docs-worker-pool/actions).